sdk/examples/xo_javascript/xo_payload.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+'use strict'
+
+const { InvalidTransaction } = require('sawtooth-sdk/processor/exceptions')
+
+class XoPayload {
+
+    fromBytes(payload) {
```

Comment:
This looks like it should be a static function that returns the promise, and the promise returns an XoPayload with those fields.